src/fileio/fileio_scen.cpp

```diff
@@ -100,20 +100,22 @@ fs::path locate_scenario(std::string scen_name) {
 	return scenPath;
 }
 
-bool load_scenario(fs::path file_to_load, cScenario& scenario, bool only_header) {
+bool load_scenario(fs::path file_to_load, cScenario& scenario, bool only_header, bool pop_paths) {
 	// Before loading a scenario, we may need to pop scenario resource paths.
-	fs::path graphics_path = ResMgr::graphics.popPath();
-	for(auto p : graphics_path) {
-		if(p.string() == "data") {
-			ResMgr::graphics.pushPath(graphics_path);
-			break;
+	if(pop_paths){
+		fs::path graphics_path = ResMgr::graphics.popPath();
+		for(auto p : graphics_path) {
+			if(p.string() == "data") {
```

Comment:
I suspect this check is _intended_ to solve the issue that you added `pop_paths` for, but I'm not sure… do you know what "data" is? Is that related to the install path of a scons build?

Reply:
`build/Blades of Exile/data` is the output folder for all platforms where the assets are stored. So I think that check is determining if the path on top of the stack is a final output path, and if it is, leaving it. That said, I haven't looked at the Resource Manager code at all, so I don't know why it has a push/pop abstraction in it to begin with. So I'm just guessing.

Reply:
The intent is to not pop the core path to the assets distributed with the game. When loading a scenario, if it has graphics or sounds inside it, those paths are pushed. When the scenario is unloaded, the paths specific to that scenario must be popped, but they might not have been pushed in the first place, so what this does is pop the top path, check if it's the core game asset path, and push it back on if it is.

Reply:
…does that make sense? It sounds rather convoluted…

Reply:
Ok, that does make sense. I forgot that there would be contextual paths for scenarios--so the pop is to remove the context of whatever scenario the player was in before.

So is it okay that I added the non-pop option? Seems cleaner than the other alternative I considered, putting the files in a data folder even though they're not being packaged for output.

Reply:
I guess it could just only pop if there are more than one, maybe? idk. I think pop_path is a fine solution.

Reply:
Hmm… I guess another option would be to put the test path _underneath_ the main data path, though that seems potentially a bit stupid (it might mean the main data path is on the stack twice?).

I'm not sure what's best here. I guess the variable isn't terrible, but I'd definitely prefer if it wasn't necessary.

Perhaps the _best_ way would be for the scenario to record whether or not it pushed paths…

Reply:
Let's at least add a TODO comment or open an issue about this, because I really don't like the use of an extra parameter here. We should find something better.

Reply:
What about this -- push_path() takes a bool argument, `temporary` or somesuch, and there's a stack of bools corresponding to the stack of paths, marking which ones are temporary and should pop.

Reply:
Hmm… I'm not too fond of that idea either, but… it might be slightly less bad than the extra parameter in the scenario loading functions.

I already mentioned what I think the best approach probably is.